dcos-log/api/v2/handlers.go

```diff
@@ -510,11 +545,65 @@ func journalHandler(w http.ResponseWriter, r *http.Request) {
 			{
 				_, err := io.Copy(w, j)
 				if err != nil {
-					logrus.Errorf("error while reading journald: %s. Request: %s", err, r.RequestURI)
+					logrus.Errorf("error while reading journald: %s. Request: %s", err, req.RequestURI)
 				}
 				f.Flush()
 			}
 		}
 	}
 
 }
+
+func browseFiles(w http.ResponseWriter, req *http.Request) {
+	r, err := setupFilesAPIReader(req, "/files/browse")
+	if err != nil {
+		e, ok := err.(errSetupFilesAPIReader)
```

Comment:
I see this pattern a lot, where you compare something to the `errSetupFilesAPIReader`. i wonder if we could break some of this out into a utility function to reduce the repetition?